test/project.t

```diff
@@ -500,17 +500,9 @@ class TestBug1900(TestCase):
 
     def test_project_eval(self):
         """1900: Project name can contain dashes"""
-        self.t("add foo project:due-b")
+        self.t("add foo project:doo-bee")
         code, out, err = self.t("_get 1.project")
-        self.assertEqual("due-b\n", out)
-
-        self.t("add foo project:scheduled-home")
-        code, out, err = self.t("_get 2.project")
-        self.assertEqual("scheduled-home\n", out)
-
-        self.t("add foo project:entry-work")
-        code, out, err = self.t("_get 3.project")
-        self.assertEqual("entry-work\n", out)
```

Comment:
While it's true that these tests test a superset of the condition "can a project name contain a dash", the special case of testing a DOM lookup of `<valid DOM ref>-<invalid DOM ref>` should be ultimately handled correctly (by failing the DOM ref lookup and the parser falling back to the raw string interpretation).

IMHO it's fine to regress on this for now, but we should move these tests into a separate test method marked with `@unittest.expectedFailure` decorator :slightly_smiling_face:

Reply:
I commented on this [here](https://github.com/GothenburgBitFactory/taskwarrior/issues/2683#issuecomment-997121680).  I'm not sure the test is representing expected behavior.  In particular, it tests that the *first* non-description attribute in `task add` does not parse DOM references, but other attributes are parsed.  I think we should consider the expected behavior in an issue, and then write tests for the result of that consideration.  I filed #2705 for that.

Reply:
I knew I saw this discussed somewhere before the holidays!